lib/std/io/tty.zig

```diff
@@ -133,3 +132,34 @@ pub const Config = union(enum) {
         };
     }
 };
+
+/// Obtains the size of a terminal designated by the file descriptor, or null if that wasn't possible,
+/// either because the file descriptor doesn't refer to a terminal or something else.
+pub fn getSize(file_descriptor: File) ?struct { rows: u16, columns: u16 } {
+    if (native_os == .windows) {
+        var info: windows.CONSOLE_SCREEN_BUFFER_INFO = undefined;
+        if (windows.kernel32.GetConsoleScreenBufferInfo(file_descriptor.handle, &info) != windows.FALSE) {
+            // In the old Windows console, info.dwSize.Y is the line count of the
+            // entire scrollback buffer, so we use this instead so that we
+            // always get the size of the screen.
+            const screen_height = info.srWindow.Bottom - info.srWindow.Top;
+            return .{
+                .rows = @intCast(screen_height),
+                .columns = @intCast(info.dwSize.X),
+            };
+        } else {
+            return null;
+        }
+    } else {
+        var winsize: posix.winsize = undefined;
+        const err = posix.system.ioctl(file_descriptor.handle, posix.T.IOCGWINSZ, @intFromPtr(&winsize));
+        if (posix.errno(err) == .SUCCESS) {
+            return .{
+                .rows = winsize.row,
+                .columns = winsize.col,
+            };
+        } else {
+            return null;
+        }
+    }
+}
```

Comment:
Would it be worth it to check the error code and establish a set of errors to return?

Reply:
All possible errors seem to be

```
EBADF
d is not a valid descriptor.
EFAULT
argp references an inaccessible memory area.
EINVAL
Request or argp is not valid.
ENOTTY
d is not associated with a character special device.
ENOTTY
The specified request does not apply to the kind of object that the descriptor d references.
```

I don't think the first three can happen. The fourth can. The fifth if it can happen can probably be treated as more or less the same thing.

This means in practice there should only be one case where failure can occur. So I think an optional probably makes sense.
I don't think this is much different for Windows' case which only gives you a boolean there.

Reply:
I think your analysis makes sense.  If we wanted to add an error here, it would probably just be `error{NotATty}`.  `EBADF` is clearly unreachable, maybe the others are as well, if not then we can use `unexpectedError`.

Reply:
True, I guess that's cleaner. Done that. Also changed the documentation to be clear that the file descriptor not referring to a terminal is really the only possible cause of failure.